macos/FluentUI/Button/Button.swift

```diff
@@ -388,125 +388,77 @@ class ButtonCell: NSButtonCell {
 			return NSZeroRect
 		}
 
-		let isRTL = NSApp.userInterfaceLayoutDirection == .rightToLeft
-		let imageRect = super.imageRect(forBounds: rect)
+		let layoutDirectionSign : CGFloat = NSApp.userInterfaceLayoutDirection == .rightToLeft ? -1 : 1
 		let titleSize = title.size(withAttributes: [.font: font as Any])
 		let imageSize = image.size
 
-		var x = imageRect.origin.x
-		var y = imageRect.origin.y
-		var width = imageRect.size.width
-		var height = imageRect.size.height
+		var x = (rect.width - imageSize.width) / 2
+		var y = (rect.height - imageSize.height) / 2
+
+		let xOffsetFromTitle = (titleSize.width + titleToImageSpacing) / 2
+		let yOffsetFromTitle = (titleSize.height + titleToImageSpacing - titleToImageVerticalSpacingAdjustment) / 2
 
 		switch imagePosition {
 		case .noImage:
-			x = 0
-			y = 0
-			width = 0
-			height = 0
+			return NSZeroRect
 		case .imageOnly, .imageOverlaps:
```

Comment:
Are the cases circled in red the imageOverlaps case? What is the scenario for that use case?

<img width="1202" alt="Screen Shot 2020-12-30 at 11 57 55 PM" src="https://user-images.githubusercontent.com/68076145/103400814-5b10b380-4afb-11eb-95bb-de8b9a3884d7.png">

Reply:
I don't think there's an explicit use case, but it's an enum value and we want to handle all the enum values.

Reply:
I covered this case and included an example to ensure the layout is as intended (image and text properly centered together).  I didn't really care that this specific example looks ugly in the test app.  If you are trying to imagine a use case where it wouldn't look ugly... it would probably be a semitransparent image or an elaborate framing/border for the title.